sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -62,10 +62,14 @@ namespace experimental {
 template <typename... Args>
 int printf(const __SYCL_CONSTANT_AS char *__format, Args... args) {
 #ifdef __SYCL_DEVICE_ONLY__
+#ifdef __SPIR__
   return __spirv_ocl_printf(__format, args...);
 #else
   return ::printf(__format, args...);
-#endif
+#endif // __SPIR__
+#else
+  return ::printf(__format, args...);
+#endif // __SYCL_DEVICE_ONLY__
```

Comment:
I think this can be simplified to:
```c++
#if defined(__SYCL_DEVICE_ONLY__) && defined(__SPIR__)
  return __spirv_ocl_printf(__format, args...);
#else
  return ::printf(__format, args...);
#endif // defined(__SYCL_DEVICE_ONLY__) && defined(__SPIR__)
```

Reply:
Yeap, done.